src/main/java/org/ecocean/servlet/export/EncounterAnnotationExportExcelFile.java

```diff
@@ -86,12 +91,12 @@ private void setMediaAssetCounts(Vector rEncounters, Shepherd myShepherd) {
       MarkedIndividual id = enc.getIndividual();
       if (id!=null && !individualIDsChecked.contains(id.getIndividualID())) {
         individualIDsChecked.add(id.getIndividualID());
-        int numNames = enc.getIndividual().numNames();
+        int numNames = enc.getIndividual().getNames().getKeys().size();
         //System.out.println("Individual "+enc.getIndividual()+" isnull = "+(enc.getIndividual()==null)+" and has # names: "+numNames);
```

Comment:
couple questions on this here:

1. was `indiv.numNames()` giving an incorrect value for some reason? if so, i would suggest we instead fix _that_ to return the expected thing. no sense having a malfunctioning method
2. if you are going for a _different_ computed value with this, can we instead move this to a **new method** on MarkedIindividual and use that here?
3. worth noting that `indiv.names` (what is returned by `getNames()`) _can_ return `null`, so there is the potential of a NPE here. so that should be handled.

Reply:
i was testing for individual `LP-B-01` Encountner catalogNumber `e591df09-3525-43d5-8cdf-8e24d6e1fb87` and find out 
enc.getIndividual().numNames() returns 5 
but enc.getIndividual().getNames().getSortedKeys() gives Array of size 4 there was mismatch thats why i updated the calculation for numNames
<img width="582" alt="Screenshot 2024-03-13 at 11 00 59 AM" src="https://github.com/WildMeOrg/Wildbook/assets/104499051/7691ce36-6c39-41ab-8263-4943f24f4edc">

Reply:
to be more clear 
"{"":[""],"Merged ":[""],"Merged":["WD00070"],"*":["LP-B-01","WD00070"]}"
this is the string value as you can see * key has 2 values so thats why numNames() returns 5 but in [writeLabel](https://github.com/WildMeOrg/Wildbook/blob/0473418ad1116f43df1653b33f0d4a1b785f2211/src/main/java/org/ecocean/servlet/export/MultiValueExportColumn.java#L54) it use [names.getValue()](https://github.com/WildMeOrg/Wildbook/blob/0473418ad1116f43df1653b33f0d4a1b785f2211/src/main/java/org/ecocean/MultiValue.java#L124) which returns only one value so for export functionality i updated this numNumes to keys size to avoid extra empty column

Reply:
> to be more clear "{"":[""],"Merged ":[""],"Merged":["WD00070"],"*":["LP-B-01","WD00070"]}" this is the string value as you can see * key has 2 values so thats why numNames() returns 5 but in [writeLabel](https://github.com/WildMeOrg/Wildbook/blob/0473418ad1116f43df1653b33f0d4a1b785f2211/src/main/java/org/ecocean/servlet/export/MultiValueExportColumn.java#L54) it use [names.getValue()](https://github.com/WildMeOrg/Wildbook/blob/0473418ad1116f43df1653b33f0d4a1b785f2211/src/main/java/org/ecocean/MultiValue.java#L124) which returns only one value so for export functionality i updated this numNumes to keys size to avoid extra empty column

ah this makes sense: `numNames()` seems to be doing the right thing, but you have a need for a different value.

looking at Names class, i wonder if you could just use `.getKeyList().size()`, as it appears that would give the right value and also (if i am looking at it correctly) handles the `null` case by returning an empty List. so either that or just catching the null case in your code should be fine.

Reply:
hi i have updated it to  getNameKeys().size(); which handle the null-case as well